File: src/pocketcloud/cloud/config/type/ConfigTypes.php
<?php

namespace pocketcloud\cloud\config\type;

use pocketcloud\cloud\util\enum\EnumTrait;

/**
 * @method static ConfigType JSON()
 * @method static ConfigType YAML()
 * @method static ConfigType PROPERTIES()
 */
final class ConfigTypes {
    use EnumTrait;

    protected static function init(): void {
        self::register("json", new JsonConfigType());
        self::register("yaml", new YamlConfigType());
        self::register("properties", new PropertiesConfigType());
    }
}

File: src/main/java/de/pocketcloud/cloud/config/type/ConfigTypes.java
package de.pocketcloud.cloud.config.type;

import de.pocketcloud.cloud.util.FileUtils;
import lombok.Getter;
import lombok.experimental.Accessors;

import java.nio.file.Files;
import java.nio.file.Path;
import java.util.List;

@Getter
@Accessors(fluent = true)
public enum ConfigTypes {

    YAML(new YamlConfigType(), List.of("yaml", "yml")),
    JSON(new JsonConfigType(), List.of("json")),
    PROPERTIES(new PropertiesConfigType(), List.of("properties")),
    ENV(new EnvironmentConfigType(), List.of("env"));

    private final ConfigType type;
    private final List<String> extensions;

    ConfigTypes(ConfigType type, List<String> extensions) {
        this.type = type;
        this.extensions = extensions;
    }

    public static ConfigType detect(Path path) {
        if (Files.isRegularFile(path)) {
            String extension = FileUtils.extensionOf(path);
            for (ConfigTypes configType : ConfigTypes.values()) {
                if (configType.extensions().contains(extension)) {
                    return configType.type();
                }
            }
        }

        return null;
    }
}